src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java

```diff
@@ -0,0 +1,650 @@
+    private static final String DEFAULT_ROW_INFLUENCE = "0";
+
+    /** Specify nearby text pattern to trigger filter to begin suppression. */
+    private Pattern nearbyTextPattern = Pattern.compile(DEFAULT_NEARBY_TEXT_PATTERN);
+
+    /**
+     * Specify check name pattern to suppress. Property can also be a RegExp group index
+     * at {@code nearbyTextPattern} in format of {@code $x} and be picked from target file.
+     */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String checkPattern = DEFAULT_CHECK_PATTERN;
+
+    /** Specify check violation message pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String messagePattern;
+
+    /** Specify check ID pattern to suppress. */
+    @XdocsPropertyType(PropertyType.PATTERN)
+    private String idPattern;
+
+    /**
+     * Specify negative/zero/positive value that defines the number of lines
+     * preceding/at/following the suppressing nearby text. Property can also be a RegExp group
+     * index at {@code nearbyTextPattern} in format of {@code $x} and be picked from target file.
+     */
+    private String rowInfluence = DEFAULT_ROW_INFLUENCE;
+
+    /**
+     * Setter to specify nearby text pattern to trigger filter to begin suppression.
+     *
+     * @param pattern a {@code Pattern} value.
+     */
+    public final void setNearbyTextPattern(Pattern pattern) {
+        nearbyTextPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check name pattern to suppress. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public final void setCheckPattern(String pattern) {
+        checkPattern = pattern;
+    }
+
+    /**
+     * Setter to specify check violation message pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setMessagePattern(String pattern) {
+        messagePattern = pattern;
+    }
+
+    /**
+     * Setter to specify check ID pattern to suppress.
+     *
+     * @param pattern a {@code String} value.
+     */
+    public void setIdPattern(String pattern) {
+        idPattern = pattern;
+    }
+
+    /**
+     * Setter to specify negative/zero/positive value that defines the number
+     * of lines preceding/at/following the suppressing nearby text. Property can also
+     * be a RegExp group index at {@code nearbyTextPattern} in
+     * format of {@code $x} and be picked from target file.
+     *
+     * @param format a {@code String} value.
+     */
+    public final void setRowInfluence(String format) {
+        rowInfluence = format;
+    }
+
+    @Override
+    public boolean accept(AuditEvent event) {
+        boolean accepted = true;
+        if (event.getViolation() != null) {
+            final FileText fileText = getFileText(event.getFileName());
+            if (fileText != null) {
```

Comment:
From what I am seeing, there is no caching here. If 1 file has 1,000 violations, you will be reloading the file 1,000 times just to check for violations. `SuppressWithNearbyCommentFilter` has some caching in it.

Reply:
Please look at https://github.com/checkstyle/checkstyle/pull/12151#discussion_r961459466 and https://github.com/checkstyle/checkstyle/commit/d565e02f4a0f00365eaadf85d005d5dbae616c80 (https://github.com/checkstyle/checkstyle/blob/d565e02f4a0f00365eaadf85d005d5dbae616c80/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyPlainTextCommentFilter.java#L427-L441 in particular). 

Initially I added a hashmap to avoid this so it is still up for discussion because of the surviving mutations. I couldn't see a way of killing them.

The optimization can still be improved by even avoiding calls to `getFileText()` but let me know if what I did back in the beginning was okay.

Reply:
Pitest doesn't mean anything that fails it should be removed. First it needs to be investigated why it is failing and what can be done to avoid the failure. For us, this usually involves regression to find cases and examining the code in detail, almost step by step. If it can't be confirmed how to avoid the failure, the next step would be to determine if there is negative impact to removing the code and having a discussion on it.

@nrmancuso Had the right idea to ignore pitest in the initial pass, and focus on functionality. That instance never said to remove the code pitest reported and my comment is showing why this was a bad idea as we will negatively impact performance.

We do need to bring back caching, and we need to examine what we can do to appease pitest AFTER we are satisfied with the functionality of the module. I would again look at `SuppressWithNearbyCommentFilter`. It has caching as well, I would look to see it fails pitest and if it doesn't, I would examine how it is able to satisfy pitest's requirement.

Reply:
I am not seeing any Done, so I assume this is still an issue.

Reply:
Done. Instead of comparing file contents as `SuppressWithNearbyCommentFilter` does, 
https://github.com/checkstyle/checkstyle/blob/d3a5f9a5ba4a1f2df28900dd70f2204c92943521/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java#L417
I am comparing the absolute paths of different files.
https://github.com/checkstyle/checkstyle/blob/22ac533ebefb7ee93f7b26aa3b186ce77b88ef4a/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilter.java#L380
Seems like a valid workaround for me.